packages/asset-swapper/src/network/chain.ts

```diff
@@ -283,8 +291,9 @@ function canBatchCallWith(ethCall: QueuedEthCall, batch: QueuedEthCall[]): boole
     if (batch.length === 0) {
         return true;
     }
+    const batchGas = batch.reduce((a, v) => a + (v.opts.gas || DEFAULT_CALL_GAS_LIMIT), 0);
     // TODO: have the sources provide realistic gas limits and split based on that.
-    if (batch.length >= 48) {
+    if ((ethCall.opts.gas || DEFAULT_CALL_GAS_LIMIT) + batchGas >= 384e6) {
```

Comment:
Tried a bunch of values and this seemed to have the best performance characteristics. Might change in the future when we add more accurate gas limits to all the other sources.